x/params/simulation/msgs.go

```diff
@@ -0,0 +1,123 @@
+package simulation
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+type simParamChange struct {
+	subspace string
+	key      string
+	subkey   string
+	simValue func(r *rand.Rand) string
+}
+
+// paramChangePool defines a static slice of possible simulated parameter changes
+// where each simParamChange corresponds to a ParamChange with a simValue
+// function to generate a simulated new value.
+//
+// NOTE: All parameter value ranges are adapted from appStateRandomizedFn.
```

Comment:
could we use a common definition for the ranges if possible (ignore if too onerous)

Reply:
Perhaps. Do you mean defining the valid ranges in the `simulation` package?

Reply:
precisely

Reply:
Yeah sure, I think that's doable.

Reply:
Updated.